### PR TITLE
I'm Yelling Timber

### PR DESCRIPTION
**Pull request by mjmagee991**
Replaced BMSLogger with EVT-core's logger
Cleaned up formatting and documentation
Pulled the latest changes from the project template

### CODE REVIEW

**Comment by mjmagee991** [comment]
[Delete BMSLogger](https://trello.com/c/rygC2ydH/48-delete-bmslogger)

File: include/dev/BQ76952.hpp [hunk 1]
```diff
@@ -237,7 +238,7 @@ class BQ76952 {
     };
 
     /**
-     * Contains a mapping between the target cell and the cooresponing
+     * Contains a mapping between the target cell and the corresponing
```
Comment (AarC10) [inline include/dev/BQ76952.hpp:241]:
```suggestion
     * Contains a mapping between the target cell and the corresponding
```

File: include/dev/SystemDetect.hpp [hunk 2]
```diff
@@ -32,7 +32,7 @@ class SystemDetect {
     };
 
     /**
-     * Create the system detect device which will work to indetify the
+     * Create the system detect device which will work to indentify the
```
Comment (AarC10) [inline include/dev/SystemDetect.hpp:35]:
```suggestion
     * Create the system detect device which will work to identify the
```

File: src/BMS.cpp [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
                                             bq(bq),
+                                            state(State::START),
                                             interlock(interlock),
                                             alarm(alarm),
-                                            systemDetect(systemDetect) {
-
-    state = State::START;
+                                            systemDetect(systemDetect),
+                                            stateChanged(true) {
     //bmsOK.writePin(EVT::core::IO::GPIO::State::LOW);
```
Comment (AarC10) [inline src/BMS.cpp:20]: Do we need this line of code here

File: src/BMS.cpp [hunk 4]
```diff
@@ -157,16 +157,16 @@ void BMS::transferSettingsState() {
 
         // If the number of errors are over the max
         if (numAttemptsMade >= MAX_BQ_COMM_ATTEMPTS) {
-            // If the settings did not transfer successfully, transiton tp
+            // If the settings did not transfer successfully, transition tp
```
Comment (AarC10) [inline src/BMS.cpp:160]:
```suggestion
            // If the settings did not transfer successfully, transition to
```

File: src/BMS.cpp [hunk 5]
```diff
@@ -279,8 +279,8 @@ void BMS::clearVoltageReadings() {
     totalVoltage = 0;
 
     // Zero out each cell voltage
-    for (uint8_t i = 0; i < DEV::BQ76952::NUM_CELLS; i++) {
-        cellVoltage[i] = 0;
+    for (uint16_t& v : cellVoltage) {
+        v = 0;
```
Comment (AarC10) [inline src/BMS.cpp:283]:
```suggestion
    memset(cellVoltage, 0, DEV::BQ76952::NUM_CELLS);
```
There are faster ways to zero out the cell voltages than using a for loop

File: src/BQSettingStorage.cpp [hunk 6]
```diff
@@ -81,17 +81,17 @@ static CO_ERR COBQSettingRead(CO_OBJ_T* obj, CO_NODE_T* node, void* buf, uint32_
...
-    BMS::LOGGER.log(BMS::BMSLogger::LogLevel::INFO, "WRITE REQUEST MADE");
-    BMS::LOGGER.log(BMS::BMSLogger::LogLevel::INFO, "Bytes incoming: %u", len);
+    log::LOGGER.log(log::Logger::LogLevel::INFO, "WRITE REQUEST MADE");
+    log::LOGGER.log(log::Logger::LogLevel::INFO, "Bytes incoming: %u", len);
 
-    uint8_t* buffer = (uint8_t*) buf;
+    auto* buffer = (uint8_t*) buf;
 
     if (len == 0)
         return CO_ERR_NONE;
```
Comment (AarC10) [inline src/BQSettingStorage.cpp:90]: Add brackets to make them consistent with the rest of the code

**Review by AarC10 — CHANGES_REQUESTED** [review]

File: include/dev/SystemDetect.hpp [hunk 7]
```diff
@@ -32,7 +32,7 @@ class SystemDetect {
...
      * provided heart beat CANopen IDs.
      *
      * @param[in] bikeHeartBeat The heart beat CANopen ID associated with the
```
Comment (DeLozM8) [inline include/dev/SystemDetect.hpp:38]:
```suggestion
     * @param[in] bikeHeartBeat The heartbeat CANopen ID associated with the
```
suggestion to fix in all areas where "heart beat" is written

File: targets/system_detect/main.cpp [hunk 8]
```diff
@@ -90,8 +92,8 @@ int main() {
     EVT::core::DEV::M24C32 eeprom(0x50, i2c);
 
     // Intialize the logger
```
Comment (DeLozM8) [inline targets/system_detect/main.cpp:94]:
```suggestion
    // Initialize the logger
```

File: src/BMS.cpp (hunk 3 above)
Comment (mjmagee991) [reply src/BMS.cpp:20]: Yes

**Review by ActuallyZach — APPROVED** [review]

**Review by AarC10 — APPROVED** [review]

**Pull request merged**